### PR TITLE
5pm-d implement basic course table

### LINKED ISSUE #56: Convert JSON format into a table of course Id, title, instructor
# User Story

As a user I can see the course titles, Id's and instructors when searching so that I know the basic information of each course in a better format than JSON.

# Acceptance Criteria

- [x] Every course title, Id, and the instructor is displayed in table format.
- [x] Courses that don't have a specified instructor still show up correctly.

# Implementation Todos

Front end:

- [x] Extract the necessary information from the JSON and display it in a table format.

Back end:

- No changes needed

Testing:

- [x] Front end tests pass and there is adequate coverage
- [x] Back end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by kunalhandaUCSB**
This pull request closes #56. This branch adds the instructor's column to the search results.

### CODE REVIEW

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js [hunk 1]
```diff
@@ -2,22 +2,78 @@ import React from "react";
...
+          title: course.title,
+          unitsFixed: course.unitsFixed
+          };
+         sections.push(section);
+       }
+      )
+    }
+  );
+const rowStyle = (row, rowIndex) => {
+  console.log(`row=`,row, `rowIndex`, rowIndex);
```
Comment (scottpchow23) [inline javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:23]: I wonder whether we still need this? Unless we're logging something that should be seen even in production, we should probably remove this.

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js [hunk 2]
```diff
@@ -2,22 +2,78 @@ import React from "react";
...
+  return (row.section % 100 == 0)? {backgroundColor: '#CEDEFA'}: (rowIndex % 2 == 1)? {backgroundColor: '#EDF3FE'}: {backgroundColor: '#FFFFFF'}
+}
+const dataAlignment = (cell, row) => {
+  // console.log(`cell=${cell} row=`, row);
```
Comment (scottpchow23) [inline javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:27]: We probably don't need this anymore.

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js [hunk 3]
```diff
@@ -2,22 +2,78 @@ import React from "react";
...
+  const alignmnet = (row.section % 100 == 0)? 'left': 'right';
+  return alignmnet
+}
+const renderCourseId = (cell, row) => {
+  // console.log(`cell=${cell} row=`, row);
```
Comment (scottpchow23) [inline javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:32]: We probably don't need this anymore.

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js [hunk 4]
```diff
@@ -2,22 +2,78 @@ import React from "react";
...
+  const courseId = (row.section % 100 == 0)? row.course.courseId: "";
+  return (  <span>{courseId}</span> )
+}
+const renderCourseTitle = (cell, row) => {
+  const courseTitle = (row.section % 100 == 0)? row.course.title: "";
+  return (  <span>{courseTitle}</span> )
+}
+const renderInstructors = (cell, row) => {
+  // console.log(`cell=${cell} row=`, row);
```
Comment (scottpchow23) [inline javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:41]: We probably don't need this anymore.

**Review by scottpchow23 — CHANGES_REQUESTED** [review]
Couple of suggestions, but this looks good on the implementation end. Add testing to this and we should be good to go!

**Comment by pconrad** [comment]
It appears that the main thing holding this back from being mergable is front end testing.  If you'd like some help with that, let me know.

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js (hunk 1 above)
Comment (hima-97) [reply javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:23]: Which line are you saying to remove for this specific PR?

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js (hunk 2 above)
Comment (hima-97) [reply javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:27]: Which line are you saying to remove for this specific PR?

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js (hunk 3 above)
Comment (hima-97) [reply javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:32]: Which line are you saying to remove for this specific PR?

File: javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js (hunk 4 above)
Comment (hima-97) [reply javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js:41]: Which line are you saying to remove for this specific PR?

**Pull request closed without merging**